mono/metadata/threads.c

```diff
@@ -1534,9 +1562,12 @@ ves_icall_System_Threading_Thread_Sleep_internal(gint32 ms)
 		mono_thread_clr_state (thread, ThreadState_WaitSleepJoin);
 
 		if (alerted) {
-			MonoException* exc = mono_thread_execute_interruption ();
-			if (exc) {
-				mono_set_pending_exception (exc);
+			MonoExceptionHandle exc = mono_thread_execute_interruption_handle (error);
```

Comment:
I'm still quite weary of adding new tech debt (`mono_thread_execute_interruption_handle` as a copy of all the logic of `mono_thread_execute_interruption`) for the sake of `Thread.Sleep_internal`.  Are we going to update all the other callers of `mono_thread_execute_interruption` to use the handle version in relatively short order?  If not, maybe we could either leave Sleep_internal out of this PR, or else change `mono_thread_execute_interruption_handle` to just be a wrapper around `mono_thread_execute_interruption` for now?
Having to maintain two copies of thread interruption in sync doesn't sound fun.

Otherwise, this LGTM.

Reply:
Yes.
I went back and forth on this.
I tried just layering pointer over handle.
That didn't work, for not understood reasons.

So I decided cloning the logic is easier/safer in the short term.

But ues it is also gross duplication.

I hope the pointer versions fall away.

I could try again though.
I could also try layering handle over pointer in the short term,
but that, arguably, puts off more work.